ezsnmp/helpers.py

```diff
@@ -23,7 +25,21 @@
 OID_INDEX_RE2 = re.compile(r"^([^\.]+::[^\.]+)\.(.*)$")
 
 
-def normalize_oid(oid, oid_index=None):
```

Comment:
Let's change this function signature to 
`normalize_oid(oid="", oid_index="")`. This should make the typing easier. 

Then the function code becomes

```python3
def normalize_oid(oid="", oid_index=""):
    """
    Ensures that the index is set correctly given an OID definition.

    :param oid: the OID to normalize
    :param oid_index: the OID index to normalize
    """

    # Determine the OID index from the OID if not specified
    if oid_index == "" and oid != "":
        first_match = OID_INDEX_RE.match(oid)
        second_match = OID_INDEX_RE2.match(oid)

        if second_match:
            oid, oid_index = second_match.group(1, 2)

        elif first_match:
            oid, oid_index = first_match.group(1, 2)

        elif oid == ".":
            oid = "."
            oid_index = ""

    return oid, oid_index
```

Reply:
NTD, I attempted this and it would cause too much change.
It'll need to be investigated further.